### PR TITLE
simpler icat timestamps

### PR DESCRIPTION
**Pull request by kaperoo**
## Description
Simplified timestamps to ISO 9075 (without milliseconds and time zones).
The `createTime`, `modTime`, `startDate` and `endDate` dates are formatted in `table.component` to avoid code repetition.
All dates are being converted to UTC time zone.

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
connect to #1605

### CODE REVIEW

File: packages/datagateway-common/src/table/table.component.tsx [hunk 1]
```diff
@@ -163,6 +163,22 @@ const VirtualizedTable = React.memo(
       disableSelectAll,
     } = props;
 
+    // Format dates to be more readable
+    // Format only if the date has a time component
+    // (to avoid formatting dates like 2020-01-01)
+    React.useEffect(() => {
```
Comment (louise-davies) [inline packages/datagateway-common/src/table/table.component.tsx:169]: This only fixes the problem in tables, and not in card view

**Review by louise-davies — CHANGES_REQUESTED** [review]

File: packages/datagateway-common/src/table/table.component.tsx (hunk 1 above)
Comment (kaperoo) [reply packages/datagateway-common/src/table/table.component.tsx:169]: oh, sorry, I completely forgot about card view 😅

**Comment by kaperoo** [comment]
Added timestamp formatting to card view.

**Review by louise-davies — APPROVED** [review]
LGTM

**Pull request merged**